ccp/src/prover.rs

```diff
@@ -109,31 +112,69 @@ impl ToCCStatus for CCProver {
 }
 
 impl CCProver {
-    pub fn new(config: CCPConfig) -> CCResult<Self> {
+    pub async fn new(config: CCPConfig) -> CCResult<Self> {
+        let msr_enforcer = MSRModeEnforcer::from_os(config.optimizations.msr_enabled);
+        let state_storage = StateStorage::new(config.state_dir.clone());
+        Self::create_prover(config, None, msr_enforcer, state_storage).await
+    }
+
+    pub async fn from_saved_state(config: CCPConfig) -> CCResult<Self> {
+        let state_storage = StateStorage::new(config.state_dir.clone());
+        let prev_state = match state_storage.try_to_load_data().await? {
+            Some(prev_state) => prev_state,
+            None => return Self::new(config).await,
+        };
+
+        let epoch = Some(prev_state.epoch_params);
+        let msr_enforcer = MSRModeEnforcer::from_preset(
+            config.optimizations.msr_enabled,
```

Comment:
IMHO we should store/read MSR opt state from the saved state.
Consider the example, CCP process crashed with  msr enabled, but right before the crash MSR opt has been disabled in the config. The next CCP start(after the failure) will cause the original MSR values to be lost b/c the original preset won't be restored.

Reply:
sounds good, will add this